devimint/src/tests.rs

```diff
@@ -1993,102 +2004,176 @@ pub async fn test_client_config_change_detection(
+    // Look for existing meta module to copy its configuration
+    let existing_meta_consensus = consensus_config_modules
+        .values()
+        .find(|module_config| module_config["kind"].as_str() == Some("meta"));
 
-        let existing_meta_consensus = existing_meta_consensus
-            .with_context(|| {
-                format!("No existing meta module found in consensus config for peer {peer_id}")
-            })?
-            .clone();
-        let existing_meta_client = existing_meta_client
-            .with_context(|| {
-                format!("No existing meta module found in client config for peer {peer_id}")
-            })?
-            .clone();
+    let existing_meta_client = client_consensus_modules
+        .values()
+        .find(|module_config| module_config["kind"].as_str() == Some("meta"));
 
-        // Find the highest existing module ID for the new module
-        let max_id = consensus_config_modules
-            .keys()
-            .chain(client_consensus_modules.keys())
-            .filter_map(|id| id.parse::<u32>().ok())
-            .max()
-            .unwrap_or(0);
+    let existing_meta_consensus = existing_meta_consensus
+        .with_context(|| {
+            format!("No existing meta module found in consensus config for peer {peer_id}")
+        })?
+        .clone();
+    let existing_meta_client = existing_meta_client
+        .with_context(|| {
+            format!("No existing meta module found in client config for peer {peer_id}")
+        })?
+        .clone();
 
-        let new_module_id = (max_id + 1).to_string();
+    // Find existing meta module in private config
+    let existing_meta_instance_id = server_config
+        .consensus
+        .modules
+        .iter()
+        .find(|(_, config)| config.kind.as_str() == "meta")
+        .map(|(id, _)| *id)
+        .with_context(|| {
+            format!("No existing meta module found in private config for peer {peer_id}")
+        })?;
+
+    let existing_meta_private = server_config
+        .private
+        .modules
+        .get(&existing_meta_instance_id)
+        .with_context(|| format!("Failed to get existing meta private config for peer {peer_id}"))?
+        .clone();
 
-        info!(
-            "Adding new meta module with id {} for peer {} (copying existing meta module config)",
-            new_module_id, peer_id
-        );
+    // Find the highest existing module ID for the new module
+    let consensus_max = consensus_config_modules
+        .keys()
+        .filter_map(|id| id.parse::<u32>().ok())
+        .max()
+        .unwrap_or(0);
+
+    let client_max = client_consensus_modules
+        .keys()
+        .filter_map(|id| id.parse::<u32>().ok())
+        .max()
+        .unwrap_or(0);
+
+    let private_max = server_config
+        .private
+        .modules
+        .keys()
+        .map(|id| u32::from(*id))
+        .max()
+        .unwrap_or(0);
```

Comment:
Which module doesn't show up in consensus config making querying other configs necessary?